pkg/server/plugin/upstreamca/awssecret/awssecret_test.go

```diff
@@ -81,37 +79,30 @@ func (as *AWSSecretSuite) TestGetSecretFail() {
 }
 
 func (as *AWSSecretSuite) Test_SubmitValidCSR() {
-	const testDataDir = "_test_data/csr_valid"
-	validCsrFiles, err := ioutil.ReadDir(testDataDir)
+	validSpiffeID := "spiffe://localhost"
+	csr, _, err := util.NewCSRTemplate(validSpiffeID)
```

Comment:
you're ignoring the public key you are returning. I was thinking something like this

```
vailidSpiffeID := "spiffe://localhost"
csr, pubKey, err := util.NewCSRTemplate(validSpiffeID)
as.Require().NoError(err)

resp, err := as.awsUpstreamCA.SubmitCSR(ctx, &upstreamca.SubmitCSRRequest{Csr: csr})
as.Require().NoError(err)
as.Require().NotNil(resp)

// Parse the leaf cert from the chain.
cert, err := x509.ParseCertificate(resp.CertChain)
as.Require().NoError(err)

as.Require().True(cryptoutil.PublicKeyEqual(cert.PublicKey, pubKey))
```

Reply:
@ZymoticB 
Thank you for your comment.
I fixed with https://github.com/spiffe/spire/pull/988/commits/91c575e4f37e37a08bd6faa347650f75788e2fe8 .